packages/standard-components/src/DataForm.svelte

```diff
@@ -33,14 +38,39 @@
   }
 
   async function save() {
+    // prevent double clicking firing multiple requests
+    if (saving) return
+    saving = true
     const SAVE_RECORD_URL = `/api/${model}/records`
     const response = await _bb.api.post(SAVE_RECORD_URL, newModel)
     const json = await response.json()
 
-    store.update(state => {
-      state[model] = state[model] ? [...state[model], json] : [json]
-      return state
-    })
+    if (response.status === 200) {
+      store.update(state => {
+        state[model] = state[model] ? [...state[model], json] : [json]
+        return state
+      })
+      
+      resetForm()
+
+      saved = true
+      setTimeout(() => {
+        saved = false
+      }, 1000)
+    }
+    saving = false
+  }
+
+  const resetForm = () => {
```

Comment:
This seems weird.. Can we not just bind the inputs to the model object, rather than binding all the state to the DOM?

Reply:
unfortunately not - svelte wont let us use `bind:value ` when the `type` of the input is dynamic